h2o-core/src/main/java/hex/Model.java

```diff
@@ -1605,6 +1613,18 @@ public Frame score(Frame fr, String destination_key, Job j, boolean computeMetri
     return output;
   }
 
+  private Frame applyTEIfModelAvailable(Frame newFr) {
+    if(_output._te_model_key != null ) {
+      Frame encodedFrame = FrameUtils.internal_applyTargetEncoder(DKV.getGet(_output._te_model_key), newFr, false);
+      Arrays.stream(encodedFrame.keys()).forEach(key -> {
```

Comment:
Converting an Array to Stream just to iterate over it introduces unnecessary overhead. A simple `fori` would do.

Reply:
agree, mainly don't like `forEach` for another reason: it allows side-effects, and I recommend streams only when  purely functional... will replace those.